unstructured_ingest/v2/processes/connectors/delta_table.py

```diff
@@ -0,0 +1,160 @@
+import json
+from dataclasses import dataclass, field
+from multiprocessing import Process
+from pathlib import Path
+from typing import Any, Optional
+
+import pandas as pd
+from pydantic import Field, Secret
+
+from unstructured_ingest.utils.dep_check import requires_dependencies
+from unstructured_ingest.v2.interfaces import (
+    AccessConfig,
+    ConnectionConfig,
+    FileData,
+    Uploader,
+    UploaderConfig,
+    UploadStager,
+    UploadStagerConfig,
+)
+from unstructured_ingest.v2.logger import logger
+from unstructured_ingest.v2.processes.connector_registry import DestinationRegistryEntry
+
+CONNECTOR_TYPE = "delta_table"
+
+
+class DeltaTableAccessConfig(AccessConfig):
+    aws_region: Optional[str] = Field(default=None, description="AWS Region")
+    aws_access_key_id: Optional[str] = Field(default=None, description="AWS Access Key Id")
+    aws_secret_access_key: Optional[str] = Field(default=None, description="AWS Secret Access Key")
+
+
+class DeltaTableConnectionConfig(ConnectionConfig):
+    access_config: Secret[DeltaTableAccessConfig] = Field(
+        default=DeltaTableAccessConfig(), validate_default=True
+    )
+    table_uri: str = Field(
+        default=None,
+        description="The path to the target folder in the S3 bucket,"
+        "formatted as s3://my-bucket/my-folder/ or local path",
+    )
+
+
+class DeltaTableUploadStagerConfig(UploadStagerConfig):
+    pass
+
+
+@dataclass
+class DeltaTableUploadStager(UploadStager):
+    upload_stager_config: DeltaTableUploadStagerConfig = field(
+        default_factory=lambda: DeltaTableUploadStagerConfig()
+    )
+
+    def run(
+        self,
+        elements_filepath: Path,
+        output_dir: Path,
+        output_filename: str,
+        **kwargs: Any,
+    ) -> Path:
+        from unstructured_ingest.utils.table import convert_to_pandas_dataframe
```

Comment:
This import can be at the top of the page because none of the dependencies in there are connector specific.

Reply:
Changed import location